src/pages/public_registration/action_accordions/PublishingAccordion.tsx

```diff
@@ -136,148 +137,160 @@ export const PublishingAccordion = ({
+              disabled={isLoading !== LoadingState.None || !registrationIsValid}
+              data-testid={dataTestId.registrationLandingPage.tasksPanel.publishButton}
+              sx={{ mt: '1rem' }}
+              color="primary"
               variant="contained"
-              data-testid={dataTestId.registrationLandingPage.tasksPanel.publishingRequestRejectButton}
-              endIcon={<CloseIcon />}
+              endIcon={<CloudUploadIcon />}
               loadingPosition="end"
-              onClick={() => updatePendingPublishingRequest('Closed')}
-              loading={isLoading === LoadingState.RejectPublishingRequest}
-              disabled={isLoading !== LoadingState.None}>
-              {t('registration.public_page.reject_publish_request')}
+              onClick={onClickPublish}
+              loading={isLoading === LoadingState.CreatePublishingREquest}>
+              {registratorRequiresApprovalForMetadataAndFiles
+                ? t('registration.public_page.tasks_panel.request_publishing')
+                : t('common.publish')}
             </LoadingButton>
-          </Box>
-        )}
-      </AccordionDetails>
-    </Accordion>
+          )}
+
+          {canHandlePublishingRequest && !hasMismatchingPublishedStatus && (
+            <Box sx={{ mt: '1rem', display: 'flex', gap: '1rem' }}>
+              <LoadingButton
+                variant="contained"
+                data-testid={dataTestId.registrationLandingPage.tasksPanel.publishingRequestAcceptButton}
+                endIcon={<CheckIcon />}
+                loadingPosition="end"
+                onClick={() => updatePendingPublishingRequest('Completed')}
+                loading={isLoading === LoadingState.ApprovePulishingRequest}
+                disabled={isLoading !== LoadingState.None || !registrationIsValid}>
+                {t('registration.public_page.approve_publish_request')}
+              </LoadingButton>
+              <LoadingButton
+                variant="contained"
+                data-testid={dataTestId.registrationLandingPage.tasksPanel.publishingRequestRejectButton}
+                endIcon={<CloseIcon />}
+                loadingPosition="end"
+                onClick={() => updatePendingPublishingRequest('Closed')}
+                loading={isLoading === LoadingState.RejectPublishingRequest}
+                disabled={isLoading !== LoadingState.None}>
+                {t('registration.public_page.reject_publish_request')}
+              </LoadingButton>
+            </Box>
+          )}
+        </AccordionDetails>
+        <Accordion>
+          <AccordionSummary sx={{ fontWeight: 700 }} expandIcon={<ExpandMoreIcon fontSize="large" />}>
+            {`${t('my_page.messages.messages')} (${ticketMessages.length})`}
+          </AccordionSummary>
+          <AccordionDetails>
+            {ticketMessages.length > 0 ? <MessageList messages={ticketMessages} /> : t('my_page.messages.no_messages')}
+          </AccordionDetails>
```

Comment:
Det er også en case hvor man ønsker å sende melding, men det kan vi kanskje ta i en egen oppgave. Per nå må man bare gå tilbake til "Oppgaver" for å kunne svare.